accepted/0000-prometheus-tls.md

```diff
@@ -0,0 +1,238 @@
+> Prometheus and exporters.
+
+# Detailed design
+
+[design]: #detailed-design
+
+## Prometheus
+
+Prometheus 2.24.0 has introduced TLS encryption as well as basic and
+client-certificate authentication [^2]. All available configuration options are
+described in official documentation [^3]. Basic authentication credentials are
+stored in the configuration file. Passwords are hashed with bcrypt.
+
+To follow current SUSE Manager security model [^4] we should consider
+implementing following default configuration:
+
+* Upgrade Prometheus to version >= 2.24.1.
+* Create and deploy CA signed Prometheus server certificate.
+* Configure `tls_server_config` using this certificate.
+* Configure basic authentication accounts to map SUSE Manager credentials.
+
+All these steps should be implemented in Salt prometheus-formula. Alternatively
+certificate generation and deployment could be moved to a new dedicated
+certificates formula.
+
+## Node exporter
+
+Node exporter supports the same configuration options as Prometheus server in
+terms of encryption and authentication [^5]. All available options are
+described in exporter-toolkit documentation [^6].
+
+To follow current SUSE Manager security model we should consider encrypting all
+the communication between node exporters and Prometheus server. Regarding
+authentication, client certificate authentication seems to be the preferred
+approach as the certificate will be anyway available on the Prometheus server.
+It would involve following configuration steps:
+
+* Create and deploy CA signed server certificates for all monitored minions.
+* Configure `tls_server_config` in node exporter using these certificates.
+* Configure `tls_config.ca_file` in Prometheus job configuration to use server
+  certificate [^7].
+* Configure `tls_config` in Prometheus job configuration to use Prometheus
+  server certificate and key files for client cert authentication.
+
+Alternatively to last point basic authentication can be configured. Please note
+that credentials are stored in Prometheus configuration in clear text. Password
+is marked as secret and not exposed via API.
+
+Prometheus service discovery will have to provide information about the
+connection scheme http or https of the endpoint.
+
+Implementation affects following components:
+
+* prometheus-formula
+* prometheus-exporters-formula
+* Prometheus Uyuni SD
+* certificates-formula (new optional formula)
+
+## Other components
+
+The intent is to roll over this kind of HTTPS support across all the official
+Prometheus exporters in the coming months and the other projects, such as
+Prometheus, Alertmanager, Pushgateway [^5]. The functionality is implemented as
+part of Prometheus Exporter Toolkit [^11] and can be added to other exporters.
+
+The TLS support for SUSE Manager distributed unofficial exporters at the time of
+writing:
+
+* Apache exporter (no TLS support)
+* PostgreSQL exporter (no TLS support)
+* Exporter Exporter (TLS support provided)
```

Comment:
Does this mean if I want to hide my TLS-enabled node exporter behind Exporter Exporter, it's not possible?

Reply:
It is possible. I've clarified the description.